vm/transaction.go

```diff
@@ -133,8 +133,8 @@ func adaptTransaction(txn core.Transaction) *Transaction {
 		if tx.Version.Uint64() == 3 {
 			tx.ResourceBounds = utils.HeapPtr(adaptResourceBounds(t.ResourceBounds))
 			tx.Tip = new(felt.Felt).SetUint64(t.Tip)
-			tx.PaymasterData = &t.PaymasterData
-			tx.AccountDeploymentData = &t.AccountDeploymentData
+			tx.PaymasterData = nilToZero(t.PaymasterData)
```

Comment:
If `PaymasterData` and `AccountDeploymentData` is `nil`, the marshaled JSON will set these 2 fields as explicit null, which is not unmarshallable on the rust side. This fix is to ensure that they're replaced by empty slices if nil.